tests/test_bot.py

```diff
@@ -1648,6 +1649,40 @@ async def post(*args, **kwargs):
             bot.callback_data_cache.clear_callback_data()
             bot.callback_data_cache.clear_callback_queries()
 
+    @pytest.mark.skipif(not TEST_WITH_OPT_DEPS, reason="Optional dependencies not installed")
+    async def test_http2_warning(self):
+        with pytest.raises(
+            PTBUserWarning, match="You set the HTTP version for the request HTTPXRequest instance"
+        ):
+            Bot("12345:ABCDE", base_url="http://", request=HTTPXRequest(http_version="2"))
+        with pytest.raises(
+            PTBUserWarning,
+            match="You set the HTTP version for the get_updates_request HTTPXRequest instance",
+        ):
+            Bot(
+                "12345:ABCDE",
+                base_url="http://",
+                get_updates_request=HTTPXRequest(http_version="2"),
+            )
+        with pytest.raises(
+            PTBUserWarning,
+            match="You set the HTTP version for the get_updates_request and request HTTPXRequest "
+            "instance",
+        ):
+            Bot(
+                "12345:ABCDE",
+                base_url="http://",
+                request=HTTPXRequest(http_version="2"),
+                get_updates_request=HTTPXRequest(http_version="2"),
+            )
+
+    @pytest.mark.skipif(
+        TEST_WITH_OPT_DEPS, reason="Only relevant if the optional dependency is not installed"
+    )
+    async def test_http2_runtime_error(self, bot):
+        with pytest.raises(RuntimeError, match=r"python-telegram-bot\[http2\]"):
+            HTTPXRequest(http_version="2")
+
```

Comment:
```suggestion
    @pytest.mark.skipif(
        not TEST_WITH_OPT_DEPS, reason="Only relevant if the optional dependency is not installed"
    )
    async def test_http2_runtime_error(self, bot):
        with pytest.raises(RuntimeError, match=r"python-telegram-bot\[http2\]"):
            HTTPXRequest(http_version="2")
```

?

Please double check that this is actually run correctly. IIRC in the workflow yaml we list all test files that contain tests that should run without optional components. To not run all bot tests in that case it would also make sense to extract this to a standalone test class - cf `test_jobqueue.py`

Reply:
no its negated, skip if opt installed

Reply:
my bad :)